File: src/main/java/com/nanaios/polygonal_tech/registries/PolygonalTechMenuTypeRegister.java
package com.nanaios.polygonal_tech.registries;

import com.nanaios.polygonal_tech.block_entity.PhotolysisMachineMk1;
import com.nanaios.polygonal_tech.menu.base.BaseMenu;
import com.nanaios.polygonal_tech.registries.impl.DeferredMenuTypeRegister;
import com.nanaios.polygonal_tech.registries.impl.MultipleMenuTypeRegister;
import net.minecraft.world.inventory.MenuType;
import net.minecraftforge.registries.RegistryObject;

public class PolygonalTechMenuTypeRegister {
    public static MultipleMenuTypeRegister MENU_TYPES = new MultipleMenuTypeRegister();
    public static DeferredMenuTypeRegister MACHINE_GUI = MENU_TYPES.create();

    public static RegistryObject<MenuType<BaseMenu<PhotolysisMachineMk1>>> PHOTOLYSIS_MACHINE_MENU;

    static {
        PHOTOLYSIS_MACHINE_MENU = MACHINE_GUI.register(PolygonalTechNamedTokens.PHOTOLYSIS_MACHINE_MK1);
    }
}

File: src/main/kotlin/com/nanaios/polygonal_tech/main/registry/PolygonalTechMenuTypeRegister.kt
package com.nanaios.polygonal_tech.main.registry

import com.nanaios.polygonal_tech.core.register.single.DeferredSingleMenuTypeRegister
import com.nanaios.polygonal_tech.main.PolygonalTech

object PolygonalTechMenuTypeRegister {
    val MENUS = DeferredSingleMenuTypeRegister(PolygonalTech.MOD_ID)
}